estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,131 @@
+            "CHECK(price > 0)",
+            "The offer price must be strictly positive.",
+        )
+    ]
+
+    # ────────────────────────────────── Base Fields ───────────────────────────────────
+    price = fields.Float("Price")
+    validity = fields.Integer("Validity (days)", default=7)
+    status = fields.Selection(
+        [("accepted", "Accepted"), ("refused", "Refused")], copy=False
+    )
+
+    # ─────────────────────────────── Relational Fields ────────────────────────────────
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+
+    # ──────────────────────────────── Computed Fields ─────────────────────────────────
+    date_deadline = fields.Date(
+        "Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+    property_type_id = fields.Many2one(
+        "estate.property.type",
+        string="Property Type",
+        related="property_id.property_type_id",
+        store=True,
+    )
+
+    # ──────────────────────────────── Compute Methods ─────────────────────────────────
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = (
+                record.create_date or fields.Date.today()
+            ) + relativedelta(days=record.validity)
+
+    # ──────────────────────────────── Inverse Methods ─────────────────────────────────
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (
+                relativedelta(
+                    record.date_deadline, (record.create_date or fields.Date.today())
+                ).days
+                + 1
+            )
+
+    # ───────────────────────────────── Action Methods ─────────────────────────────────
+    def action_accept(self):
+        for record in self:
+            if record.property_id.state == "offer_accepted":
+                raise UserError("You can only accept one offer at a time.")
+
+            if record.property_id.state in ["sold", "cancelled"]:
+                raise UserError(
+                    "You cannot accept an offer on a sold or cancelled property."
+                )
+
+            if record.property_id.state == "new":
+                raise ValidationError(
+                    "You cannot accept an offer on a new property. (New properties are not supposed to have offers.)"
+                )
+
+            if record.status == "refused":
+                raise UserError("You cannot accept a refused offer.")
+
+            for offer in record.property_id.offer_ids:
+                if offer.id != record.id:
+                    offer.status = "refused"
+
+            record.status = "accepted"
+            record.property_id.buyer_id = record.partner_id
+            record.property_id.selling_price = record.price
+            record.property_id.state = "offer_accepted"
```

Comment:
```suggestion
        property_rec = record.property_id

        # Validation checks
        if property_rec.state == "offer_accepted":
            raise UserError("You can only accept one offer at a time.")

        if property_rec.state in ("sold", "cancelled"):
            raise UserError("You cannot accept an offer on a sold or cancelled property.")

        if property_rec.state == "new":
            raise ValidationError("You cannot accept an offer on a new property.")

        if record.status == "refused":
            raise UserError("You cannot accept a refused offer.")

        # Refuse all other offers
        other_offers = property_rec.offer_ids.filtered(lambda o: o.id != record.id)
        other_offers.write({'status': 'refused'})

        # Accept this offer and update the property
        record.status = "accepted"
        property_rec.update({
            'buyer_id': record.partner_id.id,
            'selling_price': record.price,
            'state': "offer_accepted",
        })

```
Here is some improvement in code, and you can also add comments in code like this, it's a good habit.